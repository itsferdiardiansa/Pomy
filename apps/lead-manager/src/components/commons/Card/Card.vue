<template>
  <div :class="`${prefixClass}-card`">
    <slot />
  </div>
</template>
<style lang="scss" scoped>
.#{$prefixClass}-card {
  @apply bg-gray-50 rounded-lg flex flex-col;
}
</style>
