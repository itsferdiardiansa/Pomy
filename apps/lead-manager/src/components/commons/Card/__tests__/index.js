import { mount, config } from '@vue/test-utils'
import Card from '@/components/commons/Card'

describe('common/Card', () => {
  let Component
  let prefixClass

  beforeAll(async () => {
    config.renderStubDefaultSlot = true

    Component = mount(Card)

    prefixClass = `${Component.rootVM.prefixClass}-card`
  })

  afterAll(() => {
    config.renderStubDefaultSlot = false
  })

  it('will render match snapshot', () => {
    expect(Component.element).toMatchSnapshot()
  })

  it('will render children correctly', () => {
    const ContentPage = {
      components: { Card },
      template: `
        <Card>
          <div class="content"></div>
        </Card>
      `
    }

    Component = mount(ContentPage, {
      shallow: true
    })

    expect(Component.element).toMatchSnapshot()
  })
})
