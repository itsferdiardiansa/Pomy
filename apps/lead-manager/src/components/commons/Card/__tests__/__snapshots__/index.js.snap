// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common/Card will render children correctly 1`] = `
<card-stub>
  <div
    class="content"
  />
</card-stub>
`;

exports[`common/Card will render match snapshot 1`] = `
<div
  class="ss-card"
>
  
  
</div>
`;
